pkg/client/unversioned/extensions.go

```diff
@@ -130,24 +130,24 @@ func NewExtensionsOrDie(c *Config) *ExtensionsClient {
 
 func setExtensionsDefaults(config *Config) error {
 	// if experimental group is not registered, return an error
-	g, err := latest.Group("extensions")
-	if err != nil {
-		return err
+	if !extlatest.IsEnabled() {
```

Comment:
OK. So now that I see you've moved most of install into these new latest packages, I think I can better respond.

The purpose of the current structure is to let the person packaging up the binary make a conscious and explicit choice about what codecs end up in the schema. This is not possible if the install packages are imported in multiple places throughout the code.

Notice that even adding your explicit dependency, you weren't able to remove the runtime error here. Code that uses types, in my view, should *not* have a compile-time dependency on the installation of the codec for those types, because the binary may have opinions about exactly what codecs are installed, and they might be custom ones, not ones in added by the default install package. @caesarxuchao thinks I need to give better justification for this position, so I'll try to remember the original reason for this separation and get back with you.

But the general principle I'm fighting for here is that our package import tree is too big--we should allow every use case we target to import the minimal amount of packages.